enclave_manager/avalon_enclave_manager/base_enclave_manager.py

```diff
@@ -0,0 +1,268 @@
+    """
+    try:
+        json.loads(wo_request)
+    except ValueError as e:
+        logger.error("Invalid JSON format found for workorder - %s", e)
+        return False
+    return True
+
+# -----------------------------------------------------------------
+
+
+def create_json_worker(enclave_data, config):
+    """
+    Create JSON worker object which gets saved in KvStorage
+    """
+    worker_type_data = dict()
+    worker_type_data["verificationKey"] = enclave_data.verifying_key
+    worker_type_data["extendedMeasurements"] = \
+        enclave_data.extended_measurements
+    worker_type_data["proofDataType"] = enclave_data.proof_data_type
+    worker_type_data["proofData"] = enclave_data.proof_data
+    worker_type_data["encryptionKey"] = enclave_data.encryption_key
+    worker_type_data["encryptionKeySignature"] = \
+        enclave_data.encryption_key_signature
+
+    worker_info = dict()
+    worker_info["workerType"] = WorkerType.TEE_SGX.value
+    worker_info["organizationId"] = \
+        config.get("WorkerConfig")["OrganizationId"]
+    worker_info["applicationTypeId"] = \
+        config.get("WorkerConfig")["ApplicationTypeId"]
+
+    details_info = dict()
+    details_info["workOrderSyncUri"] = \
+        config.get("WorkerConfig")["WorkOrderSyncUri"]
+    details_info["workOrderAsyncUri"] = \
+        config.get("WorkerConfig")["WorkOrderAsyncUri"]
+    details_info["workOrderPullUri"] = \
+        config.get("WorkerConfig")["WorkOrderPullUri"]
+    details_info["workOrderNotifyUri"] = \
+        config.get("WorkerConfig")["WorkOrderNotifyUri"]
+    details_info["receiptInvocationUri"] = \
+        config.get("WorkerConfig")["ReceiptInvocationUri"]
+    details_info["workOrderInvocationAddress"] = config.get(
+        "WorkerConfig")["WorkOrderInvocationAddress"]
+    details_info["receiptInvocationAddress"] = config.get(
+        "WorkerConfig")["ReceiptInvocationAddress"]
+    details_info["fromAddress"] = config.get("WorkerConfig")["FromAddress"]
+    details_info["hashingAlgorithm"] = \
+        config.get("WorkerConfig")["HashingAlgorithm"]
+    details_info["signingAlgorithm"] = \
+        config.get("WorkerConfig")["SigningAlgorithm"]
+    details_info["keyEncryptionAlgorithm"] = \
+        config.get("WorkerConfig")["KeyEncryptionAlgorithm"]
+    details_info["dataEncryptionAlgorithm"] = \
+        config.get("WorkerConfig")["DataEncryptionAlgorithm"]
+    details_info["workOrderPayloadFormats"] = \
+        config.get("WorkerConfig")["workOrderPayloadFormats"]
+    details_info["workerTypeData"] = worker_type_data
+
+    worker_info["details"] = details_info
+    worker_info["status"] = WorkerStatus.ACTIVE.value
+
+    # JSON serialize worker_info
+    json_worker_info = json.dumps(worker_info)
+    logger.info("JSON serialized worker info is %s", json_worker_info)
+    return json_worker_info
+
+# -----------------------------------------------------------------
+
+
+def bind_zmq_socket(zmq_port):
```

Comment:
Might be good to add description for thsi function -> "Zmq is used for synchronous work order processing"